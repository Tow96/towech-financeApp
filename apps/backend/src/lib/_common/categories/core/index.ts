export * from './category-aggregate';
export * from './category-events';
export * from './i-category-repository';
export * from './subcategory-entity';
export * from './subcategory-events';
