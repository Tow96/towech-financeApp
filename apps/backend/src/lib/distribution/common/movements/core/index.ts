export * from './category.value-object';
export * from './i-movement-repository';
export * from './movement-aggregate';
export * from './summary-item.value-object';
export * as events from './movement-events';
