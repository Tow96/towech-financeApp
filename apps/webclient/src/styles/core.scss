// The theming functions are heavily taken from angular material, but in a lighter way
// Original code: https://material.angular.io/
@use 'sass:map';

@function get-palette-hue($palette, $hue) {
  @return map.get($palette, $hue);
}

@function get-palette-hue-contrast($palette, $hue) {
  @return map.get(map.get($palette, contrast), $hue);
}
