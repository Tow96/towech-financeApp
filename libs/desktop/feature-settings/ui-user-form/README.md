# desktop-feature-settings-ui-user-form

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `nx test desktop-feature-settings-ui-user-form` to execute the unit tests.
