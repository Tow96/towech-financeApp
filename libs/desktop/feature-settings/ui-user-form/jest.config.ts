/* eslint-disable */
export default {
  displayName: 'desktop-feature-settings-ui-user-form',
  preset: '../../../../jest.preset.js',
  setupFilesAfterEnv: ['<rootDir>/src/test-setup.ts'],
  coverageDirectory: '../../../../coverage/libs/desktop/feature-settings/ui-user-form',
  transform: {
    '^.+\\.(ts|mjs|js|html)$': [
      'jest-preset-angular',
      {
        tsconfig: '<rootDir>/tsconfig.spec.json',
        stringifyContentPathRegex: '\\.(html|svg)$',
      },
    ],
  },
  transformIgnorePatterns: ['node_modules/(?!.*\\.mjs$)'],
  snapshotSerializers: [
    'jest-preset-angular/build/serializers/no-ng-attributes',
    'jest-preset-angular/build/serializers/ng-snapshot',
    'jest-preset-angular/build/serializers/html-comment',
  ],
  collectCoverageFrom: ['**/*.ts', '!**/index.ts', '!**/jest.config.ts'],
  coverageReporters: ['html', 'text', 'text-summary', 'cobertura', 'lcov', 'json'],
};
