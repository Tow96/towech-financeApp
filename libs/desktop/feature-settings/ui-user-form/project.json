{
  "name": "desktop-feature-settings-ui-user-form",
  "$schema": "../../../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/desktop/feature-settings/ui-user-form/src",
  "prefix": "finance",
  "tags": [],
  "projectType": "library",
  "targets": {
    "test": {
      "executor": "@nx/jest:jest",
      "outputs": ["{workspaceRoot}/coverage/{projectRoot}"],
      "options": {
        "jestConfig": "libs/desktop/feature-settings/ui-user-form/jest.config.ts",
        "passWithNoTests": true
      },
      "configurations": {
        "ci": {
          "ci": true,
          "codeCoverage": true
        }
      }
    },
    "lint": {
      "executor": "@nx/linter:eslint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": [
          "libs/desktop/feature-settings/ui-user-form/**/*.ts",
          "libs/desktop/feature-settings/ui-user-form/**/*.html"
        ]
      }
    }
  }
}
