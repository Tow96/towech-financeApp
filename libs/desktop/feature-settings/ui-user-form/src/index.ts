export * from './lib/ui-user-form.component';
