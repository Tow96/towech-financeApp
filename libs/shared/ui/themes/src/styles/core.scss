// The theming functions are heavily taken from angular material, but in a lighter way
// Original code: https://material.angular.io/
@use 'sass:map';

@function get-palette-hue($palette, $hue) {
  @return map.get($palette, $hue);
}

@function get-palette-hue-contrast($palette, $hue) {
  @return map.get(map.get($palette, contrast), $hue);
}

@function define-theme(
  $background,
  $text,
  $primary,
  $accent,
  $success: null,
  $warning: null,
  $error: null
) {
  @if $background == null or $text == null {
    @error 'Background and text colors need to be defined';
  }

  @if $primary == null or $accent == null {
    @error 'primary and accent colors need to be defined palettes';
  }

  @if not map.get($primary, contrast) {
    @error 'Primary is not a palette';
  }

  @if not map.get($accent, contrast) {
    @error 'Accent is not a palette';
  }

  @if $success != null and not map.get($success, contrast) {
    @error 'Success is not a palette';
  }

  @if $error != null and not map.get($error, contrast) {
    @error 'Error is not a palette';
  }

  @if $warning != null and not map.get($warning, contrast) {
    @error 'Warning is not a palette';
  }

  $result: (
    color: (
      text: $text,
      background: $background,
      primary: $primary,
      accent: $accent,
      success: if($success != null, $success, $accent),
      warning: if($warning != null, $warning, $accent),
      error: if($error != null, $error, $accent),
    ),
  );

  @return $result;
}

@mixin color($theme) {
  $colors: map.get($theme, color);
  $primary: map.get($colors, primary);

  :root {
    background-color: map.get($colors, background);
    color: map.get($colors, text);
  }

  div.container {
    background-color: get-palette-hue($primary, 500);
  }
}
