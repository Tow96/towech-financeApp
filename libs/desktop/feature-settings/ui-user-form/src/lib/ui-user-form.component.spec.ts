// Libraries
import { CUSTOM_ELEMENTS_SCHEMA } from '@angular/core';
import { ComponentFixture, TestBed } from '@angular/core/testing';
import { By } from '@angular/platform-browser';
import { subscribeSpyTo } from '@hirez_io/observer-spy';
// Tested elements
import { DesktopSettingsUiUserFormComponent } from './ui-user-form.component';
// Components
import { DesktopInputComponent } from '@finance/desktop/shared/ui-input';

describe('DesktopSettingsUiUserForm', () => {
  let component: DesktopSettingsUiUserFormComponent;
  let fixture: ComponentFixture<DesktopSettingsUiUserFormComponent>;
  let compiled: HTMLElement;

  const getElement = (testid: string) =>
    fixture.debugElement.query(By.css(`[data-test-id="${testid}"]`));
  const detectChanges = () => {
    fixture.debugElement.triggerEventHandler('click', null);
    fixture.detectChanges();
  };
  const setInputValue = (name: string, value: string) => {
    const input: DesktopInputComponent = getElement(name).componentInstance;
    input.writeValue(value);
    input.onChange({ target: { value } } as any);
    detectChanges();
  };

  beforeEach(() => {
    TestBed.overrideComponent(DesktopSettingsUiUserFormComponent, {
      set: { host: { '(click)': 'dummy' }, providers: [], schemas: [CUSTOM_ELEMENTS_SCHEMA] },
    });
    fixture = TestBed.createComponent(DesktopSettingsUiUserFormComponent);
    component = fixture.componentInstance;

    detectChanges();
    compiled = fixture.nativeElement;
  });

  it('Should be defined', () => expect(component).toBeTruthy());
  it('Must match the snapshot', () => expect(compiled).toMatchSnapshot());
  it('Should update the name when the input changes', () => {
    const newValue = 'Name';
    setInputValue('name', newValue);
    expect(component.form.value.name).toBe(newValue);
  });
  it('Should update the email when the input changes', () => {
    const newValue = 'mail@provider.com';
    setInputValue('mail', newValue);
    expect(component.form.value.mail).toBe(newValue);
  });
  it('Should emit when submiited', () => {
    const spy = subscribeSpyTo(component.submitted);
    const btn: HTMLElement = getElement('submit-edit-user').nativeElement.querySelector('button');
    btn.click();
    expect(spy.receivedNext()).toBeTruthy();
  });
});
