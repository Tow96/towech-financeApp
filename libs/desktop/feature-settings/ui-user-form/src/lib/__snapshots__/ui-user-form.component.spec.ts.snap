// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DesktopSettingsUiUserForm Must match the snapshot 1`] = `
<div
  id="root1"
>
  <h2>
    Edit User
  </h2>
  <form
    class="ng-untouched ng-pristine ng-valid"
    novalidate=""
  >
    <div
      class="form-input"
    >
      <p>
        Name:
      </p>
      <finance-input
        class="ng-untouched ng-pristine ng-valid"
        data-test-id="name"
        formcontrolname="name"
      >
        <div
          class="input_component"
        >
          <input
            placeholder=" "
            type="text"
          />
          <label>
            
          </label>
        </div>
      </finance-input>
    </div>
    <div
      class="form-input"
    >
      <p>
        E-Mail
      </p>
      <finance-input
        class="ng-untouched ng-pristine ng-valid"
        data-test-id="mail"
        formcontrolname="mail"
      >
        <div
          class="input_component"
        >
          <input
            placeholder=" "
            type="text"
          />
          <label>
            
          </label>
        </div>
      </finance-input>
    </div>
    <div
      class="form-submit"
    >
      <finance-button
        data-test-id="submit-edit-user"
        type="submit"
      >
        <button
          class="accent"
          type="submit"
        >
          Save
        </button>
      </finance-button>
    </div>
  </form>
</div>
`;
